src/calculation/CostElement.ts

```diff
@@ -1,23 +1,38 @@
+import { AmountKind } from '~/models/AmountKind';
 import { Cost } from '~/models/Cost';
 import { Currency } from '~/models/Currency';
 import { VatRate } from '~/models/VatRate';
 
 import { roundSum } from './roundSum';
 
 export class CostElement implements Cost {
-  readonly currency: Currency = Currency.PLN;
-
   get grossPrice(): number {
-    return this.netPrice + this.vatSum;
+    return this.amountKind === AmountKind.GROSS
+      ? this.amount
+      : this.amount + this.vatSum;
+  }
+
+  get netPrice(): number {
+    return this.amountKind === AmountKind.NET
+      ? this.amount
+      : this.amount - this.vatSum;
   }
 
   get vatSum(): number {
-    return roundSum((this.vatRate / 100) * this.netPrice);
+    if (this.amountKind === AmountKind.NET) {
+      return roundSum((this.vatRate / 100) * this.amount);
+    } else {
+      return roundSum(
+        (this.amount * (this.vatRate / 100)) / (1 + this.vatRate / 100)
+      );
+    }
   }
 
   constructor(
-    public readonly name: string,
```

Comment:
Do you think Cost/SaleElement `name` property is not needed? It seems useful to give such item a name :smile:

Reply:
I thought wrong about the CostElement as a whole invoice. Indeed, It should have a name even required. I will fix it :+1:

Reply:
After discussion: we will create a new Invoice type in the future task, this will become irrelevant.